### PR TITLE
add launchd configuration to support brew services

### PR DESCRIPTION
**Pull request by benesch**
Add a launchd configuration so that brew services can manage a
single-node Cockroach cluster, if so desired.

### CODE REVIEW

**Comment by bdarnell** [comment]
> After reading that issue, I'm going to hold off on this until we have a less-contentious HTTP port. @bdarnell do you have thoughts here? Perhaps 26256, one below the default Cockroach port?

We're not going to get a second officially-allocated port, so whatever we do will just be squatting on some unclaimed port. 26256 is as good as any. 

@tamird Is there any update on the feasibility of merging the HTTP port back onto the GRPC port? The use of 8080 was supposed to be temporary. 

> We also need to decide if letting brew services start cockroach launch an insecure cluster is acceptable.

What do other brew services do? This is opt-in and not just a consequence of `brew install`, right? If it's opt-in I'm slightly more OK with it as long as it's restricted to localhost-only (although it would be even better if we could generate a set of keys the first time the service is started)

**Comment by tamird** [comment]
> @tamird Is there any update on the feasibility of merging the HTTP port back onto the GRPC port? The use of 8080 was supposed to be temporary.

Well, the grpc fork at https://github.com/bradfitz/grpc-go *only* supports the ServeHTTP path (this is the one that would let us merge the ports back) but I'm quite sure there hasn't been any performance work on mainline to make that any more feasible. See https://github.com/cockroachdb/rpc-bench for slightly dated results (I can update if you'd like).

TL;DR there's no update, but it's still something we hope to do.

**Comment by bdarnell** [comment]
Let's rerun the benchmarks and ping https://github.com/grpc/grpc-go/issues/586 again. Managing the two ports is going to be an ongoing headache for deployments so it would be great if we could resolve it before 1.0.

**Comment by tamird** [comment]
@bdarnell done.

**Comment by benesch** [comment]
This is inexplicably stuck on an old commit. Revised by #6.
